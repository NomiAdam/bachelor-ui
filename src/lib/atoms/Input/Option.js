// eslint-disable-next-line no-unused-vars
import React from 'react';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const StyledOption = styled.option`
 	position: absolute;
	padding: 20px 2%;
    width: 96%;
    display: block;
    margin: 0px;
    cursor: pointer;
    background-color: transparent;
    font-size: 16px;
    transition: .5s all ease;
    &:hover {
    	background-color: ${ lightTheme.whiteBlue };
    };
`;

export default StyledOption;
