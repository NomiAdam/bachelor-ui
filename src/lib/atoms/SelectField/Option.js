import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyledOption = styled.div`
    width: 100%;
    display: inline-block;
    border: none;
	  padding: 10px 20px;
	  text-align: center;
	  background-color: ${ ({ selected, ...props }) => (selected ? resolveTheme(TYPES.BACKGROUND_COLOR)(props) : 'initial') };
    &:hover {
        background-color: ${ resolveTheme(TYPES.BACKGROUND_COLOR_HOVER) };
        cursor: pointer;
    }
`;

const Option = ({
    value, onClick, label, selected, secondary,
}) => (
    <StyledOption secondary={ secondary } onMouseDown={ onClick(value) } selected={ selected }>
        {label}
    </StyledOption>
);

Option.propTypes = {
    /**
     * String value of Option
     */
    value: PropTypes.any,
    /**
     * onMouseDown event handler, (value) => (e) =>
     */
    onClick: PropTypes.func.isRequired,
    /**
     * String representation of displayed value
     */
    label: PropTypes.string,
    /**
     * Boolean representation of currently selected item
     */
    selected: PropTypes.bool,
};

export { StyledOption };
export default Option;
