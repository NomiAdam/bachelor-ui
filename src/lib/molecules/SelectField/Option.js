import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const StyledOption = styled.div`
    width: 100%;
    display: inline-block;
    border: none;
	padding: 10px 20px;
	text-align: center;
	background-color: ${ ({ selected }) => (selected ? lightTheme.lightGrey : 'initial') };
    &:hover {
        background-color: ${ lightTheme.lightGrey };
        cursor: pointer;
    }
`;

const Option = ({
    value, onClick, label, selected,
}) => (
    <StyledOption onMouseDown={ onClick(value) } selected={ selected }>
        {label}
    </StyledOption>
);

Option.propTypes = {
    /**
     * String value of Option
     */
    value: PropTypes.any,
    /**
     * onMouseDown event handler, (value) => (e) =>
     */
    onClick: PropTypes.func.isRequired,
    /**
     * String representation of displayed value
     */
    label: PropTypes.string,
    /**
     * Boolean representation of currently selected item
     */
    selected: PropTypes.bool,
};

export { StyledOption };
export default Option;
