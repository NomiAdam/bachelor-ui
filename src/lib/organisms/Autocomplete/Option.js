import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {lightTheme} from '../../constants/theme';

const StyledOption = styled.div`
    width: 100%;
    border: none;
	padding: 10px 20px;
	background-color: ${({selected}) => selected ? lightTheme.lightGrey : 'initial'};
    &:hover {
        background-color: ${lightTheme.lightGrey};
        cursor: pointer;
    }
`;

const Option = ({value, onClick, selected}) => (
    <StyledOption onMouseDown={onClick(value)} selected={selected}>
        {value}
    </StyledOption>
);

Option.propTypes = {
    /**
     * String value of Option
     */
    value: PropTypes.string,
    /**
     * onMouseDown event handler, (value) => (e) =>
     */
    onClick: PropTypes.func.isRequired,
    /**
     * Boolean value whether current option is selected or not
     */
    selected: PropTypes.bool,
};

export {StyledOption};
export default Option;
