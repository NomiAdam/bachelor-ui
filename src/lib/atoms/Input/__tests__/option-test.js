/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import Option from '../Option';

describe('Option', () => {
	it('should be defined', () => {
		expect(shallow(<Option/>)).toBeDefined();
	});
	it('should match the snapshot', () => {
		expect(shallow(<Option/>)).toMatchSnapshot();
	});
});