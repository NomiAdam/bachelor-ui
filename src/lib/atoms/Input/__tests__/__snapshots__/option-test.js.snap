// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should match the snapshot 1`] = `
<option
  className="sc-bdVaJa gMFflD"
/>
`;
